### ISSUE TITLE
[Feature] AMQPNetLite EventBus transport

### ISSUE THREAD

**Issue by rwestonbutler** [dotnet-architecture/eShopOnContainers#505]
We have a situation where we are not able to use RabbitMQ or any Azure services, but we do have an AMQP channel setup on IBM Websphere. Given that the AMQPNetLite library has come along nicely, would anyone be interested in developing a transport for the event bus using this library? Thanks!

**Comment by ClouditTogether** [comment]
Hi my name is Davide,

I have been monitoring this project. The reason is simple. I have been
looking to build an idea up and this may form a good framework for  the
initial bootstrap. I have not yet had time to do anything on git but joined
this to understand the team, the status of this project and whether I can
leverage it with some peoples help.

If anyone in the team has been working on this for a while and can help me
understand the general status , this would be good. I would be happy to
chat over the phone if that's easier.

Are you asking if we can implement Rabbit or Azure service bus? Nservicebus?


Kind Regards
Davide De Vita

Mobile: 07526 963657
Email: ClouditTogether@gmail.com <Cloudittogether@gmail.com>
*www.ClouditTogether.io <http://www.CloudItTogether.io>*
https://www.linkedin.com/in/davidedevita/

**Comment by rwestonbutler** [comment]
I believe there are already EventBus implementations using Rabbit and Azure Service bus in the eShopOnContainers project. To clarify my question, what we'd be looking for is an AMQP broker agnostic implementation of an event bus using the [AMQPNetLite](https://github.com/Azure/amqpnetlite) project as a base for the transport.  Our company has a heavy investment in IBM infrastructure, and we don't see that changing anytime soon, unfortunately, but we were able to get the powers that be create an AMQP channel for testing.

**Comment by ClouditTogether** [comment]
Sorry, you are looking for someone to implement the  AMQPNetLite
<https://github.com/Azure/amqpnetlite> lib into this project?

**Comment by CESARDELATORRE** [comment]
@rwestonbutler - That’s right. eShopOnContainers provides sample implementations for RabbitMQ and Azure Service Bus, already. Although these implementations (reusable building blocks) are just samples that might not be enough for your specific case and would need to be evolved depending on your specific needs in a production environment. That is why we also recommend to explore service buses like NServiceBus, MassTransit, etc. Our implementation is only for microservices approaches learning, initially. 
eShopOnContainers is not a supported Product for production because in some cases there are cases that might not be covered, I want to make that clear.
The important point for us is to show the concepts of an event bus and eventual consistency between microservices.

About a new implementation of the event bus by using any additional infrastructure like interoperability with IBM infrastructure, we’d be happy if anyone in the community wants to create it and provide a tested PR to be added to the project, of course.

**Comment by rwestonbutler** [comment]
@CESARDELATORRE  Understood, thanks for the quick response! I appreciate all the work being done on this project, as it has been a great learning source for microservices dev patterns. I ended up here for just that reason - our team is splitting a monolith into several services, so we're doing some due diligence and POCs. I've started a POC for a bus impl using AMQPNetLite, since it can work with any AMQP broker, so maybe i can share that here if it turns out to be worthwile. 

From all my research, I've not found a bus that is broker agnostic.. searching Google for 'C# AMQP event bus' has RabbitMQ in almost all the search results, so it's been a bit frustrating! 

Again - thanks for your quick response!
